public/pages/utils/anomalyResultUtils.ts

```diff
@@ -124,32 +124,42 @@ export const prepareDataForChart = (
   if (anomalies.length > MAX_DATA_POINTS) {
     anomalies = sampleMaxAnomalyGrade(anomalies);
   }
-
-  anomalies.push({
-    startTime: dateRange.startDate,
-    endTime: dateRange.startDate,
-    plotTime: interval
-      ? dateRange.startDate - MIN_IN_MILLI_SECS * interval
-      : dateRange.startDate,
-    confidence: null,
-    anomalyGrade: null,
-  });
-  anomalies.unshift({
-    startTime: dateRange.endDate,
-    endTime: dateRange.endDate,
-    plotTime: interval
-      ? dateRange.endDate + MIN_IN_MILLI_SECS * interval
-      : dateRange.endDate,
-    confidence: null,
-    anomalyGrade: null,
-  });
-  if (getFloorPlotTime) {
+  if (getFloorPlotTime) { // we need to get floor plot time for bar chart
     anomalies = anomalies.map(anomaly => {
       return {
         ...anomaly,
         plotTime: getFloorPlotTime(anomaly.plotTime),
       };
     });
+    anomalies.push({
+      startTime: dateRange.startDate,
+      endTime: dateRange.startDate,
+      plotTime: anomalies[anomalies.length - 1].plotTime - MIN_IN_MILLI_SECS * interval,
+      confidence: null,
+      anomalyGrade: null,
+    });
+    anomalies.unshift({
+      startTime: dateRange.endDate,
+      endTime: dateRange.endDate,
+      plotTime: anomalies[0].plotTime + MIN_IN_MILLI_SECS * interval,
+      confidence: null,
+      anomalyGrade: null,
+    });
+  } else {
+    anomalies.push({
+      startTime: dateRange.startDate,
+      endTime: dateRange.startDate,
+      plotTime: dateRange.startDate,
+      confidence: null,
+      anomalyGrade: null,
+    });
+    anomalies.unshift({
+      startTime: dateRange.endDate,
+      endTime: dateRange.endDate,
+      plotTime: dateRange.endDate,
+      confidence: null,
+      anomalyGrade: null,
+    });
```

Comment:
will X Axis automatically shrink if only start/end data points are inserted? I recall I got such issue if plotting data only has start/end data.

Reply:
nvm. just realized you insert real data in above code

Reply:
Good question, let me test locally.

Reply:
I think for your case it should be fine. Anomalies can only be empty when detector is not running, and in that case, we hide live chart.